### PR TITLE
integration tests

### PR DESCRIPTION
**Pull request by Tianwei-She**
# Pull Request Summary

## Bundles
For each of (cloudpickle, creating from dirs):
1. Create a bundle
2. Create a bundle with the same name - this should error out

## Endpoints
async endpoints:
1. Create an endpoint from a bundle
2. Wait until it's initialized
3. Run a single test inference
4. Edit the endpoint - currently only the bundle
5. Delete the endpoint

## Batch jobs
1. Create a (small) batch job and run on some inputs.

## TODO
1. test sync endpoint as well when it's ready to be tested - for now it's hard to check if a sync endpoint is ready or not, so testing it can fail easily
2. edit more endpoint params
3. check the result of batch job?

About the API key:
- Loui created [a new scale account scale-launch-integration-tests@scale.com](https://dashboard.scale.com/corp/lookup/62bc820451dbea002b1c5421) thanks Loui!
- I added the API KEY as [an environment variable in CircleCI](https://app.circleci.com/settings/project/github/scaleapi/launch-python-client/environment-variables?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fscaleapi%2Flaunch-python-client%2F304%2Fworkflows%2F970ec2b0-ec65-4bdc-9e03-b0ef58d7849c%2Fjobs%2F307) 
- to run this test locally, you should add the API KEY to your environment variables
- the API KEY is saved as `launch_test_api_key` in AWS Secrets Manager

About the client initialization:
- 2 model bundles `test-returns-1` and `test-returns-2` were manually created beforehand, so that when testing endpoints there are existing bundles to use and testing endpoints won't be dependent on bundle creation
- they are defined in `conftest.py` for record (probably there's better ways to save them)

## Test Plan

CircleCI tests passed

## Shortcut Ticket

[384106](https://app.shortcut.com/scaleai/story/384106/integration-test-suite-for-launch)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+
+@pytest.fixture()
+def client():
+    # to run this test locally, you should add the API KEY to your environment variables
+    launch_test_api_key = os.getenv("LAUNCH_TEST_API_KEY")
+    client = launch.LaunchClient(api_key=launch_test_api_key)
+
+    # these two model bundles are used in test_endpoints.py
```
Comment (yixu34) [inline tests/conftest.py:55]: Not sure how this comment relates to the commented out `client.create_model_bundle` calls 🤔

**Review by yixu34 — COMMENTED** [review]
I was imagining that the integration tests would live in `models`. It's true that the tests are ofc driven by the client, and also ofc that the client talks to the server, but: the client library will eventually be public (technically is now, but we'll be hiding it again in a bit), and I don't know if we'd want (integratinon test) API calls to be driven from there. And I guess it's just an overall bias towards making sure our server APIs are working.

In the integration test, we can check out the latest version of the client library and just test that. We can additionally test a pinned version of the client too.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import logging
+import pytest
+import os
+
+import launch
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
```
[the remark this answers is not in the document]
Comment (yixu34) [reply tests/conftest.py]: Oh yeah fair - in that case we can skip it. *But*, we were talking about moving this to the internal repo anyway, in which case this is viable?

[the remark this answers is not in the document]
Comment (yixu34) [reply tests/conftest.py]: Wait I'm confused then - if we are moving the integration tests to models, then... this PR can just go away right?

**Comment by Tianwei-She** [comment]
this is moved to the `modes` repo https://github.com/scaleapi/models/pull/3603

**Pull request closed without merging**